core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/upgrade/upgrade.yml

```diff
@@ -308,33 +308,155 @@
+        name: "{{ _new_pg_service }}"
         enabled: true
+      vars:
+        _new_pg_service: "{{ new_version.pg.service_name[ansible_os_family] }}"
 
     - name: PostgreSQL | Extensions | repmgr | RedHat | Update repmgr symlink
+      when:
+        - ansible_os_family == 'RedHat'
+        - pg_is_repmgr_used
       file:
         src: "{{ new_version.repmgr.bin_dir.RedHat }}/repmgr"
         dest: "/usr/bin/repmgr"
         state: link
+
+    - name: PostgreSQL | Extensions | repmgr | Debian | Update REPMGRD_CONF in /etc/default/repmgrd
       when:
+        - ansible_os_family == 'Debian'
         - pg_is_repmgr_used
-        - ansible_os_family == 'RedHat'
+      replace:
+        path: /etc/default/repmgrd
+        regexp: ^REPMGRD_CONF=\"/etc/postgresql/{{ old_version.pg.version }}/main/repmgr.conf\"
+        replace: REPMGRD_CONF="/etc/postgresql/{{ new_version.pg.version }}/main/repmgr.conf"
 
     # Start and enable repmgr service
     - include_tasks: roles/postgresql/tasks/upgrade/extensions/replication/pg-new/repmgr-start-and-enable-service.yml
       when: pg_is_repmgr_used
 
   # Step: Reverting to old cluster
   rescue:
-    - name: PostgreSQL | Rollback | Remove PG {{ new_version.pg.version }} related packages
+    - name: Rollback | Get service facts
+      service_facts:
+
+    - name: Rollback | Extensions | repmgr | Stop all repmgr services
+      when: ansible_facts.services[service_name + '.service'] is defined
+      systemd:
+        name: "{{ service_name }}"
+        state: stopped
+      loop_control:
+        loop_var: service_name
+      loop: "{{ _services | unique }}"
+      vars:
+        _services:
+          - "{{ old_version.repmgr.service_name[ansible_os_family] }}"
+          - "{{ new_version.repmgr.service_name[ansible_os_family] }}"
+
+    - name: Rollback | Extensions | repmgr | Disable {{ _service }} service
+      when:
+        - _service != old_version.repmgr.service_name[ansible_os_family]
+        - ansible_facts.services[_service + '.service'] is defined
+      systemd:
+        name: "{{ _service }}"
+        enabled: false
+      vars:
+        _service: "{{ new_version.repmgr.service_name[ansible_os_family] }}"
+
+    - name: Restore repmgr symlinks
+      when: pg_is_repmgr_used
+      block:
+        - name: Rollback | Extensions | repmgr | RedHat | Update repmgr symlink
+          file:
+            src: "{{ old_version.repmgr.bin_dir.RedHat }}/repmgr"
+            dest: "/usr/bin/repmgr"
+            state: link
+          when: ansible_os_family == 'RedHat'
+
+        - name: Rollback | Extensions | repmgr | Debian | Update symlink to config file
+          file:
+            src: "{{ old_version.repmgr.config_dir[ansible_os_family] }}/repmgr.conf"
+            dest: /etc/repmgr.conf
+            state: link
+          when: ansible_os_family == 'Debian'
+
+    - name: Stop and disable new PostgreSQL service
+      block:
```

Comment:
Block seems to be unnecessary for one task or the next one can be moved into this block.